### PR TITLE
Use not healthy as error msg

### PR DESCRIPTION
**Pull request by cody-wang-cb**
**Description**
Currently when the backends all fail, it return `no backend is available`, which is quite unclear for the end user. Making a change here to use `no backend is currently healthy to serve traffic` instead which is a lot more clear.

**Tests**
Fixed the impacted tests

**Additional context**

Add any other context about the problem you're solving.

**Metadata**

- Fixes #[Link to Issue]

### CODE REVIEW

**Review by jelias2 — APPROVED** [review]
I agree that the returned message could be improved to include more info, so I'll approve the PR. 

However I am hesitant that this message implies that the backed is not healthy, where as  the requests could fail for additional reasons besides just backend health. Thus this error may be slightly misleading. 

Would like to get another set of eyes before merging

**Review by felipe-op — APPROVED** [review]

**Pull request merged**
